pylint/extensions/_check_docs_utils.py

```diff
@@ -164,10 +164,17 @@ def possible_exc_types(node):
                     and isinstance(val, (astroid.Instance, nodes.ClassDef))
                     and utils.inherit_from_std_ex(val)
                 ):
-                    excs.append(val.name)
+                    excs.append(val)
 
     try:
-        return {exc for exc in excs if not utils.node_ignores_exception(node, exc)}
+        final_exceptions = set()
+        for exc in excs:
+            if utils.node_ignores_exception(node, exc.name):
+                continue
+            if not isinstance(exc, nodes.ClassDef):
+                exc = exc.getattr("__class__")[0]
+            final_exceptions.add(exc)
+        return final_exceptions
     except astroid.InferenceError:
```

Comment:
This try-except statement is only needed for `utils.node_ignores_exception` because it is trying to infer the handler of the exception, see https://github.com/PyCQA/pylint/commit/5f048faf002f2af8de11ae50e3567ce872569f1a

I think we should only include that if statement in the try-except and leaving the others out. A comment explaining why this can raise `InferenceError` also wouldn't hurt.